### ISSUE TITLE
Some empty vehicles don't despawn in cargo terminals

### ISSUE THREAD

**Issue by Brechto** [CitiesSkylinesMods/TMPE#1745]
After the last update (Hotels), some trucks arriving in cargo terminals don't despawn after emptying which clogs them up so they don't accept new vehicles. Happens in all types of cargo terminals.

![Screenshot 2023-05-23 173756](https://github.com/CitiesSkylinesMods/TMPE/assets/90205622/bc912842-d303-4aa0-a7fd-b7b517d6389a)

**Comment by Brechto** [comment]
Cargo warehouse seems to work fine, but I only tested it in a new city. I also tried the cargo train terminal, which had no problems so it may just be old vehicles from the previous version?

**Comment by Brechto** [comment]
Yup it only affects existing vehicles, after clearing traffic new trucks despawn as intended.

**Comment by krzychu124** [comment]
Could be, also maybe some mod is still/was not compatible? Personally I haven't noticed anything like that when I was testing it few days ago via early access to prepare for update, also I didn't notice any significant change in the game code which could break TM:PE. I think, like you already noticed, it might be some weird bug in vanilla 🤷‍♂️

**Comment by krzychu124** [comment]
I've tested some vanilla savegames and it seems to be vanilla issue, depends on where vehicle spawned (source building) and most likely where it delivered the cargo

**Comment by Neto-Zeme** [comment]
I can confirm this issue. After a despawn it works first but after some time the issue appears again. I also noticed trucks driving out and in again into the terminal.
